src/main/scala/com/johnsnowlabs/ml/tensorflow/TensorflowBert.scala

```diff
@@ -21,36 +23,42 @@ class TensorflowBert(val tensorflow: TensorflowWrapper,
       tokens ++
       Array(sentenceEndTokenId) ++
       Array.fill(maxSentenceLength - tokens.length - 2)(0)
-
   }
 
   def tag(batch: Seq[Array[Int]], embeddingsKey: String): Seq[Array[Array[Float]]] = {
     val tensors = new TensorResources()
+    val buf = tensors.createIntBuffer(batch.length*maxSentenceLength)
+    val shape = Array(batch.length.toLong, maxSentenceLength)
 
-    //println(s"shape = ${batch.length}, ${batch(0).length}")
-    val shrink = batch.map { sentence =>
+    batch.map { sentence =>
       if (sentence.length > maxSentenceLength) {
-        sentence.take(maxSentenceLength - 1) ++ Array(sentenceEndTokenId)
+        buf.put(sentence.take(maxSentenceLength - 1) ++ Array(sentenceEndTokenId))
       }
       else {
-        sentence
+        buf.put(sentence)
       }
-    }.toArray
+    }
+    buf.flip()
+
+    val runner = tensorflow.getSession(configProtoBytes = configProtoBytes).runner
 
-    val calculated = tensorflow.getSession(configProtoBytes = configProtoBytes).runner
-      .feed(tokenIdsKey, tensors.createTensor(shrink))
+    runner
+      .feed(tokenIdsKey, tensors.createIntBufferTensor(shape, buf))
       .fetch(embeddingsKey)
-      .run()
 
-    tensors.clearTensors()
+    val outs = runner.run().asScala
+    val embeddings = TensorResources.extractFloats(outs.head)
 
-    val embeddings = TensorResources.extractFloats(calculated.get(0))
+    tensors.clearSession(outs)
+    tensors.clearTensors()
+    buf.clear()
```

Comment:
nice!!

Reply:
Yes, the `tensors.clearSession` is the part we get back all the memory! The improvement of going from +40G memory to stay at 8G over 100K sentences.